### PR TITLE
[#61] Exclude Flutter generated files form git

### PR DESCRIPTION
**Pull request by chornerman**
#61 

## What happened 👀

We need to exclude Flutter generated files from git

## Insight 📝

- [x] Add the path of files that should be excluded in `.gitignore`
- [x] Remove already included files from this repository
- [x] Add CI step to generate Flutter files before analyzing error

## Proof Of Work 📹

App and CI/CD run normally

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,9 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+
+# Flutter generated files
+*.g.dart
+*.gen.dart
+*.config.dart
+*.freezed.dart
```
Comment (luongvo) [inline .gitignore:52]: I think you will need to ignore `*.mocks.dart` too

File: .github/workflows/code_analysis_and_tests.yml [hunk 2]
```diff
@@ -25,6 +25,9 @@ jobs:
       - name: Get Flutter dependencies
         run: flutter pub get
 
+      - name: Generate Flutter files
+        run: flutter packages pub run build_runner build
```
Comment (luongvo) [inline .github/workflows/code_analysis_and_tests.yml:29]: We should run this command with `--delete-conflicting-outputs` param.

```suggestion
        run: flutter packages pub run build_runner build --delete-conflicting-outputs`
```

**Review by luongvo — COMMENTED** [review]
rest lgtm

**Review by markgravity — APPROVED** [review]
Please resolve @luongvo comments, then all good to me 🙏

**Review by luongvo — APPROVED** [review]
lgtm

**Pull request merged**